Automattic/vip-go-mu-plugins
# Alert only if more than 1000 items was autohealed

## Description
Alert for auto-healing only if the number of auto-healed records is more than 1000. Due to the async nature of indexing documents to ElasticSearch, it is quite common to be few documents in behind at any given point. This change should help us notice the more suspicious cases.

## Changelog Description

### Plugin Updated: Enterprise Search

Enhancement of internal logging for Enterprise Search.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
1) Add 2 debugging lines around `class-healthjob.php:157`:
```
		$results = $this->health->validate_index_posts_content( [ 'silent' => true ] );

                var_dump($results); // <--- HERE
                
		if ( is_wp_error( $results ) ) {
			$message = sprintf( 'Cron validate-contents error for site %d (%s): %s', FILES_CLIENT_SITE_ID, home_url(), $results->get_error_message() );
			$this->send_alert( '#vip-go-es-alerts', $message, 2 );
		} else if ( ! empty( $results ) && count( $results ) > self::AUTOHEALED_ALERT_THRESHOLD ) {
		
                        var_dump('alerting!!'); // <--- HERE
                        
			$message = sprintf( 'Cron validate-contents for site %d (%s): Autohealing executed for %d records.', FILES_CLIENT_SITE_ID, home_url(), count( $results ) );
			$this->send_alert( '#vip-go-es-alerts', $message, 2 );
		}
```
2) Make the validate content always enabled `class-healthjob.php:308` :
```
public function is_enabled() {
		return true;
```
3) Introduce inconsistency by changing a document in ES
4) Run following and see that we get the inconsitency but no alert:
```
dev-env exec -- wp eval "do_action('vip_search_health_validate_content');"
```
